### PR TITLE
Database Suitability TOPP Tool

### PR DESCRIPTION
**Pull request by Waschi97**
- db suitability score, ms2-id-rate & tsv output

- test for the new tool

### CODE REVIEW

File: src/topp/DatabaseSuitability.cpp [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+/**
+@page TOPP_DatabaseSuitability DatabaseSuitability
+
+@brief Calculates a suitability for a database which was used a for peptide identification search. Also reports the quality of LC-MS spectra.
+
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:60]: missing `@endcond` at the very end. See other TOPP tools

File: src/topp/DatabaseSuitability.cpp [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+class DatabaseSuitability :
+  public TOPPBase
+{
+public:
+  DatabaseSuitability() :
+    TOPPBase("DatabaseSuitability", "Computes a suitability score for a database which was used for a peptide identification search. Also reports the quality of LC-MS spectra.", false)
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:67]: you can add a citation and cite the original paper.
Maybe also do this in the class description above.

File: src/topp/DatabaseSuitability.cpp [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+protected:
+
+  // this function will be used to register the tool parameters
+  // it gets automatically called on tool execution
+  void registerOptionsAndFlags_() override
+  {
+    registerInputFile_("in_id", "<file>", "", "Input idXML file from peptide search (after FDR)");
+    setValidFormats_("in_id", { "idXML" });
+    registerInputFile_("in_spec", "<file>", "", "Input MzML file");
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:79]: is this the mzML used for peptide id? :)

File: src/topp/DatabaseSuitability.cpp [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+  {
+  }
+
+protected:
+
+  // this function will be used to register the tool parameters
+  // it gets automatically called on tool execution
+  void registerOptionsAndFlags_() override
+  {
+    registerInputFile_("in_id", "<file>", "", "Input idXML file from peptide search (after FDR)");
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:77]: is this the combined DBEngine + DeNovoEngine search?
What Engines are supported? (also add to main docu)

File: src/topp/DatabaseSuitability.cpp [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+    setValidFormats_("in_id", { "idXML" });
+    registerInputFile_("in_spec", "<file>", "", "Input MzML file");
+    setValidFormats_("in_spec", { "mzML" });
+    registerInputFile_("in_novo", "<file>", "", "Input idXML file containing de novo peptides");
+    setValidFormats_("in_novo", { "idXML" });
+    registerOutputFile_("out", "<file>", "", "Optional tsv output", false); //tsv output
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:83]: What does this contain. i can see it's `tsv` format from the specs. Provide some useful information here.
E.g what columns does it contain?

File: src/topp/DatabaseSuitability.cpp [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+    setValidFormats_("out", { "tsv" });
+    registerIntOption_("novor_fract", "<integer>", 1, "Set the percentage of de novo peptides to capture with a score higher than the fasta score.", false, true);
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:85]: not sure what the description means...
and fractions in OpenMS are usually given as 0.0 - 1.0. (set a valid range to enforce this)

File: src/topp/DatabaseSuitability.cpp [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+    registerFlag_("force_no_re_rank", "Use this flag if you want to disable re-ranking. This might yeild in underperformance.", true);
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:86]: what is re-ranking? Short description here and longer description in the docs.
typo in `yeild`, but `underestimate quality` is probably more accurate?

File: src/topp/DatabaseSuitability.cpp [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+
+    vector<ProteinIdentification> novo_prots;
+    vector<PeptideIdentification> novo_peps;
+    x.load(in_novo, novo_prots, novo_peps);
+
+    UInt64 count_ms2_lvl;
+    {
+      MzMLFile m;
+      PeakFileOptions op;
+      op.addMSLevel(2);
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:119]: not what you wanted probably.
`setMSLevels`?

File: src/topp/DatabaseSuitability.cpp [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    UInt64 count_db = 0;
+    UInt64 count_novo = 0;
+    UInt64 count_re_ranked = 0;
+    UInt64 count_interest = 0;
+
+    for (const auto& pep_id : pep_ids)
+    {
+      vector<PeptideHit> hits = pep_id.getHits();
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:152]: no copy, take by const ref. Also for other assignments below

File: src/topp/DatabaseSuitability.cpp [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+        {
+          ++count_interest;
+          // check for re-ranking
+          if (no_re_rank)
+          {
+            ++count_novo;
+            continue;
+          }
+
+          if (double(top_hit.getMetaValue("MS:1002252")) - double(second_hit.getMetaValue("MS:1002252")) <= cut_off)
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:214]: the metavalue might not exist... and this will crash hard

File: src/topp/DatabaseSuitability.cpp [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
+    OPENMS_LOG_INFO << "MS2 ID rate: " << id_rate << endl << endl;
+
+    if (!out.empty())
+    {
+      OPENMS_LOG_INFO << "Writting output to: " << out << endl << endl;
+
+      std::ofstream os(out);
+      os.precision(writtenDigits(double()));
+      os << "#top_db_hits\t" << count_db << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:259]: add a header...

File: src/topp/DatabaseSuitability.cpp [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+      os << "#top_novo_hits\t" << count_novo << endl;
+      os << "db_suitability\t" << quality << endl;
+      os << "#total_novo_seqs\t" << count_novo_seq << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:262]: do not use `endl` unless you want to flush to console/disk at this very moment. Otherwise its just slow

File: src/topp/DatabaseSuitability.cpp [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+    UInt curr_hit = 1;
+
+    for (const auto& hit : pep_id.getHits())
+    {
+      if (curr_hit > 10) break;
+      ++curr_hit;
+
+      if (!hit.metaValueExists("target_decoy"))
+      {
+        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run before hand."));
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:291]: FDR needs to run...

File: src/topp/DatabaseSuitability.cpp [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+  }
+
+private:
+
+  double getDecoyDiff_(const PeptideIdentification& pep_id)
+  {
+    double diff = -1;
+
+    // get the score of the first two decoy hits
+    double decoy_1 = -1;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:280]: using `-1` for all of these is a bit dangerous on the off-chance that a score is -1 for some (unknown) search engine.
You could switch to Infitinity or some other extreme (numeric_limits<double>::max()) ....

File: src/topp/DatabaseSuitability.cpp [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+      os.close();
+    }
+
+    return EXECUTION_OK;
+
+  }
+
+private:
+
+  double getDecoyDiff_(const PeptideIdentification& pep_id)
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:275]: add documentation to all functions

File: src/topp/DatabaseSuitability.cpp [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
+      }
+    }
+
+    if (double(diffs.size()) / pep_ids.size() < 0.2)
+    {
+      throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Under 20 % of peptide identifications have two decoy hits. This is not enough for re-ranking. Use the 'force_no_re_rank' flag to still compute a suitability score."));
+    }
+
+    // sort the diffs decreasing
+    std::sort(diffs.begin(), diffs.end(), std::greater<double>());
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:341]: partial_sort is enough?

Comment (cbielow) [reply src/topp/DatabaseSuitability.cpp:341]: simply sort the first `novor_fract`*N of N positions (or `(1-novor_fract)*N in reverse order to save on sorting effort). And then grab the diff at that very position.
No need for the two loops below!?

File: src/topp/DatabaseSuitability.cpp [hunk 17]
```diff
@@ -0,0 +1,367 @@
...
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+        {
+          is_novo = false;
+          break;
+        }
+      }
+
+      if (is_novo) // top hit is de novo hit
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:172]: deal with DB-case here first, and omit the large if-block for deNovo...
less indentations for the reader

File: src/topp/DatabaseSuitability.cpp [hunk 18]
```diff
@@ -0,0 +1,367 @@
...
+      PeptideHit top_hit = hits[0];
+
+      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
+
+      // check if top hit is found in de novo protein
+      set<String> accessions = top_hit.extractProteinAccessionsSet();
+      bool is_novo = true;
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:165]: technically, the hit could be shared between DB and deNovo...
And should then be counted as DB, right?

File: src/topp/DatabaseSuitability.cpp [hunk 19]
```diff
@@ -0,0 +1,367 @@
...
+      {
+        if (hits.size() == 1)
+        {
+          ++count_novo;
+          continue;
+        }
+
+        PeptideHit second_hit = hits[1];
+
+        if (second_hit.getMetaValue("target_decoy") == "decoy")
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:182]: should we then just ignore decoy hits and look for the next target? (except if its the top hit, as above. then we skip over the whole PSM)

File: src/topp/DatabaseSuitability.cpp [hunk 20]
```diff
@@ -0,0 +1,367 @@
...
+        {
+          ++count_novo;
+          continue;
+        }
+
+        // check if second hit is db hit
+        set<String> second_accessions = top_hit.extractProteinAccessionsSet();
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:189]: this code is duplicated. Make it a function or a class, which gets passed a PeptideHit (or even better the PepID) and computes everything you want to know.

File: src/topp/DatabaseSuitability.cpp [hunk 21]
```diff
@@ -0,0 +1,367 @@
...
+    OPENMS_LOG_INFO << count_interest << " times scored a de novo hit just above a database hit. Of those times " << count_re_ranked << " top de novo hits where re-ranked using a decoy cut-off of " << cut_off << endl;
+    OPENMS_LOG_INFO << "Database suitability: " << quality << endl << endl;
+    OPENMS_LOG_INFO << unique_novo.size() << " unique de novo sequences" << endl;
+    OPENMS_LOG_INFO << count_novo_seq << " total de novo sequences" << endl; 
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
+    OPENMS_LOG_INFO << "MS2 ID rate: " << id_rate << endl << endl;
+
+    if (!out.empty())
+    {
+      OPENMS_LOG_INFO << "Writting output to: " << out << endl << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:255]: `Writing`

File: src/topp/DatabaseSuitability.cpp [hunk 22]
```diff
@@ -0,0 +1,367 @@
...
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
+    OPENMS_LOG_INFO << count_novo << " top hits that were only found in the concatenated de novo peptide." << endl;
+    OPENMS_LOG_INFO << count_interest << " times scored a de novo hit just above a database hit. Of those times " << count_re_ranked << " top de novo hits where re-ranked using a decoy cut-off of " << cut_off << endl;
+    OPENMS_LOG_INFO << "Database suitability: " << quality << endl << endl;
+    OPENMS_LOG_INFO << unique_novo.size() << " unique de novo sequences" << endl;
+    OPENMS_LOG_INFO << count_novo_seq << " total de novo sequences" << endl; 
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
+    OPENMS_LOG_INFO << "MS2 ID rate: " << id_rate << endl << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:251]: not the classical way to interpret ID rate. Mention that this only counts deNovo IDs

File: src/topp/DatabaseSuitability.cpp [hunk 23]
```diff
@@ -0,0 +1,367 @@
...
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    double quality = double(count_db) / (count_db + count_novo);
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
+    OPENMS_LOG_INFO << count_novo << " top hits that were only found in the concatenated de novo peptide." << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:245]: currently, its not `only`. But if you fix the check above that should be fine.

File: src/topp/DatabaseSuitability.cpp (hunk 13 above)
Comment (jpfeuffer) [reply src/topp/DatabaseSuitability.cpp:291]: Not for this meta value though, right? PeptideIndexer should be enough for that.
If you need to ensure that FDR ran, you need to check for a q-value/FDR score (check IDScoreSwitcher findScoreType functionality).

File: src/topp/DatabaseSuitability.cpp [hunk 24]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    double quality = double(count_db) / (count_db + count_novo);
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:244]: maybe add the total number of identified MS2 spectra, e.g. `123 / 456 (30%) of all hits were found using the database.`
same for deNovo

File: src/topp/DatabaseSuitability.cpp [hunk 25]
```diff
@@ -0,0 +1,367 @@
...
+    OPENMS_LOG_INFO << count_novo << " top hits that were only found in the concatenated de novo peptide." << endl;
+    OPENMS_LOG_INFO << count_interest << " times scored a de novo hit just above a database hit. Of those times " << count_re_ranked << " top de novo hits where re-ranked using a decoy cut-off of " << cut_off << endl;
+    OPENMS_LOG_INFO << "Database suitability: " << quality << endl << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:247]: `Database suitability (score in [0, 1]): ...` ??

File: src/topp/DatabaseSuitability.cpp [hunk 26]
```diff
@@ -0,0 +1,367 @@
...
+    OPENMS_LOG_INFO << unique_novo.size() << " unique de novo sequences" << endl;
+    OPENMS_LOG_INFO << count_novo_seq << " total de novo sequences" << endl; 
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:250]: name number of total MS2 spectra and of identified MS2 spectra (deNovo + DB)

File: src/topp/DatabaseSuitability.cpp (hunk 18 above)
Comment (Waschi97) [reply src/topp/DatabaseSuitability.cpp:165]: yes and this should happen here right?
`is_novo` will only stay 'true' as long as each accessions contains `CONCAT_PEPTIDE` and if this isn't found in at least one accession the hit will be counted as DB

File: src/openms/source/APPLICATIONS/ToolHandler.cpp [hunk 27]
```diff
@@ -53,6 +53,7 @@ namespace OpenMS
     tools_map["ConsensusMapNormalizer"] = Internal::ToolDescription("ConsensusMapNormalizer", "Map Alignment");
     tools_map["CometAdapter"] = Internal::ToolDescription("CometAdapter", "Identification");
     tools_map["CruxAdapter"] = Internal::ToolDescription("CruxAdapter", "Identification");
+    tools_map["DatabaseSuitability"] = Internal::ToolDescription("DatabaseSuitability", "Quality Control");
```
Comment (cbielow) [inline src/openms/source/APPLICATIONS/ToolHandler.cpp:56]: add the tool to the doxygen page here:
https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_documentation.html

maybe a new section `Quality Control` is suitable now, since we can also move the QC tool there (currently at `Misc`).

File: src/topp/DatabaseSuitability.cpp (hunk 18 above)
Comment (cbielow) [reply src/topp/DatabaseSuitability.cpp:165]: ahh... right. my bad.

File: src/topp/DatabaseSuitability.cpp (hunk 19 above)
Comment (Waschi97) [reply src/topp/DatabaseSuitability.cpp:182]: My thought process was, if the first hit is a target-novo hit and the second one is a decoy hit it wouldn't make sense to consider re-ranking even if the third hit would be a target-db hit. But technically those could of cause still score close to each other...

File: src/topp/DatabaseSuitability.cpp (hunk 10 above)
Comment (Waschi97) [reply src/topp/DatabaseSuitability.cpp:214]: While calculating the decoy cut-off all hits are checked to have this meta-value. I thought it wouldn't be necessary to check for existence, but I guess it would be safer to still check..

File: src/topp/DatabaseSuitability.cpp (hunk 18 above)
Comment (jpfeuffer) [reply src/topp/DatabaseSuitability.cpp:165]: Another thing to consider: peptides can (rarely) map to both target and decoy proteins yielding "target+decoy" (I think) for the meta-value. How would you handle those? Most of the time handling them as targets is fine but you should keep it in mind.

File: src/topp/DatabaseSuitability.cpp [hunk 28]
```diff
@@ -64,7 +65,7 @@ class DatabaseSuitability :
 {
 public:
   DatabaseSuitability() :
-    TOPPBase("DatabaseSuitability", "Computes a suitability score for a database which was used for a peptide identification search. Also reports the quality of LC-MS spectra.", false)
```
Comment (jpfeuffer) [inline src/topp/DatabaseSuitability.cpp]: You can actually specify the citation in the constructor, such that it is correctly formatted and printed when running it on the command line.

File: src/topp/DatabaseSuitability.cpp [hunk 29]
```diff
@@ -149,55 +152,46 @@ class DatabaseSuitability :
...
         }
 
-        PeptideHit second_hit = hits[1];
-
-        if (second_hit.getMetaValue("target_decoy") == "decoy")
+        // find the second target hit, skip all decoy hits inbetween
+        const PeptideHit* second_hit = nullptr;
+        for (Int i = 1; i < hits.size(); ++i)
+        {
+          if (hits[i].getMetaValue("target_decoy") == "target")
```
Comment (jpfeuffer) [inline src/topp/DatabaseSuitability.cpp:181]: Hm as I mentioned. Here it probably would be better to do something like:
startsWith("target") in case you have a "target+decoy" peptide.

Comment (Waschi97) [reply src/topp/DatabaseSuitability.cpp:181]: Yes, this will come in the next commit! :)

File: src/topp/DatabaseSuitability.cpp (hunk 28 above)
Comment (jpfeuffer) [reply src/topp/DatabaseSuitability.cpp]: (as a separate parameter to the constructor, I mean)

File: src/topp/DatabaseSuitability.cpp [hunk 30]
```diff
@@ -0,0 +1,372 @@
...
+    IdXMLFile x;
+    vector<ProteinIdentification> prot_ids;
+    vector<PeptideIdentification> pep_ids;
+    x.load(in_id, prot_ids, pep_ids);
+
+    vector<ProteinIdentification> novo_prots;
+    vector<PeptideIdentification> novo_peps;
+    x.load(in_novo, novo_prots, novo_peps);
+
+    UInt64 count_ms2_lvl;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:117]: `size_t` or `Size` is the "native" type of `exp.size()`

File: src/topp/DatabaseSuitability.cpp [hunk 31]
```diff
@@ -0,0 +1,372 @@
...
+    }
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+
+    // db suitability
+
+
+    double cut_off;
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:143]: value of cut_off is undefined .. and might never be defined...

File: src/topp/DatabaseSuitability.cpp [hunk 32]
```diff
@@ -0,0 +1,372 @@
...
+      // skip if the top hit is a decoy hit
+      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
+
+      // check if top hit is found in de novo protein
+      if (!isNovoHit_(top_hit)) // top hit is db hit
+      {
+        ++count_db;
+        continue;
+      }
+      else // top hit is novo hit
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:176]: no need for else ...{} if using continue above

File: src/topp/DatabaseSuitability.cpp [hunk 33]
```diff
@@ -0,0 +1,372 @@
...
+          continue;
+        }
+
+        // find the second target hit, skip all decoy hits inbetween
+        const PeptideHit* second_hit = nullptr;
+        for (UInt i = 1; i < hits.size(); ++i)
+        {
+          if (hits[i].getMetaValue("target_decoy") == "target")
+          {
+            second_hit = &hits[i];
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:191]: break afterwards? otherwise you might compare 1st vs 3rd

File: src/topp/DatabaseSuitability.cpp [hunk 34]
```diff
@@ -0,0 +1,372 @@
...
+    auto sort_end = diffs.begin() + (1 - novor_fract) * diffs.size();
+    
+    std::partial_sort(diffs.begin(), sort_end + 1, diffs.end(), std::greater<double>());
+
+    return *sort_end;
+  }
+
+  bool isNovoHit_(const PeptideHit& hit)
+  {
+    set<String> accessions = hit.extractProteinAccessionsSet();
```
Comment (cbielow) [inline src/topp/DatabaseSuitability.cpp:367]: can be `const set<String> accesssions = `
just to avoid the copy if `hit.extractProteinAccessionsSet()` returns an l-value (in the future)

File: src/topp/DatabaseSuitability.cpp (hunk 30 above)
Comment (cbielow) [reply src/topp/DatabaseSuitability.cpp:125]: also for all the other UInt64 occurrences

File: doc/doxygen/Doxyfile.in [hunk 35]
```diff
@@ -2457,7 +2457,7 @@ INTERACTIVE_SVG        = NO
 # found. If left blank, it is assumed the dot tool can be found in the path.
 # This tag requires that the tag HAVE_DOT is set to YES.
 
-DOT_PATH               =
```
Comment (jpfeuffer) [inline doc/doxygen/Doxyfile.in:2460]: Hmm.. I think it would be better to leave it blank or configure it with the basename of \@DOXYGEN_DOT_EXECUTABLE\@

**Comment by jpfeuffer** [comment]
As a reference from our Discord discussion. Very cool.

File: doc/doxygen/Doxyfile.in (hunk 35 above)
Comment (Waschi97) [reply doc/doxygen/Doxyfile.in:2460]: whoops - yes of cause!

**Comment by jpfeuffer** [comment]
Looks good. Some minor concerns: 
- Can the filesizes be reduced easily? E.g. filtering by a small massrange? Since we do not really benchmark the results here, it would not need to be a full-sized input.
- Did you add some handled/non-exception-throwing irregularities? E.g. for my tools I sometimes add some empty PeptideIdentifications or some weird scores, or a target+decoy peptide, so that other people see how those cases are currently expected to be handled. I think this could also be addressed later when writing class tests if it is too hard to add to a full dataset.

**Comment by jpfeuffer** [comment]
Ok so we figured that it is not very easy to reduce size and have meaningful results. The file also actually includes some corner cases like "target+decoy" entries.
We will address other corner cases in future class tests after benchmarking.
